bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/client/XmlRpcClient.java

```diff
@@ -111,39 +105,18 @@ protected String getRpcCallbackUrl() {
             }
             Request req = httpClient.POST(url).content(content).timeout(config.getTimeout(), TimeUnit.SECONDS)
                     .header(HttpHeader.CONTENT_TYPE, "text/xml;charset=" + config.getEncoding());
-            try {
-                ret = req.send().getContent();
-            } catch (IllegalArgumentException e) { // Returned buffer too large
-                logger.info("Blocking XmlRpcRequest failed: {}, trying non-blocking request", e.getMessage());
-                InputStreamResponseListener respListener = new InputStreamResponseListener();
-                req.send(respListener);
-                Response resp = respListener.get(config.getTimeout(), TimeUnit.SECONDS);
-                ByteArrayOutputStream respData = new ByteArrayOutputStream(RESP_BUFFER_SIZE);
-
-                int httpStatus = resp.getStatus();
-                if (httpStatus == HttpStatus.OK_200) {
-                    byte[] recvBuffer = new byte[RESP_BUFFER_SIZE];
-                    try (InputStream input = respListener.getInputStream()) {
-                        while (true) {
-                            int read = input.read(recvBuffer);
-                            if (read == -1) {
-                                break;
-                            }
-                            respData.write(recvBuffer, 0, read);
-                        }
-                        ret = respData.toByteArray();
-                    }
-                } else {
-                    logger.warn("Non-blocking XmlRpcRequest failed, status code: {} / request: {}", httpStatus,
-                            request);
-                    resp.abort(new IOException());
-                }
+            FutureResponseListener listener = new FutureResponseListener(req, config.getBufferSize() * 1024);
+            req.send(listener);
+            ContentResponse response = listener.get(config.getTimeout(), TimeUnit.SECONDS);
+            ret = response.getContent();
+            if (ret == null || ret.length == 0) {
+                throw new IOException("Received no data from the Homematic gateway");
             }
             if (logger.isTraceEnabled()) {
                 String result = new String(ret, config.getEncoding());
                 logger.trace("Client XmlRpcResponse (port {}):\n{}", port, result);
             }
-        } catch (UnsupportedEncodingException | ExecutionException | TimeoutException | InterruptedException e) {
+        } catch (Exception e) {
```

Comment:
Please keep the catch explicit as it was before. I'd rather not have RuntimeExceptions get treated as IOExceptions.

Reply:
OK, accepted. 

I have changed the catch clause and explicitly added IllegalArgumentException because this is throws by Jetty if the buffer is too small.

Reply:
If the buffer is too small, then why is `sendMessage` retrying the request when that happens?

Reply:
I agree, that it does not really make sense, but this would mean some additional changed in another place and I wanted to avoid to it together with these changes. It's on my to-do list for a larger refactoring.